tapir/log/views.py

```diff
@@ -50,3 +59,80 @@ def create_text_log_entry(request, **kwargs):
 
     form.save()
     return safe_redirect(request.GET.get("next"), "/", request)
+
+
+class LogTable(django_tables2.Table):
+    entry = django_tables2.Column(
+        empty_values=(), accessor="as_leaf_class__render", verbose_name="Message"
+    )
+
+    class Meta:
+        model = LogEntry
+        template_name = "django_tables2/bootstrap.html"
+        fields = ["created_date", "actor", "user", "entry"]
+
+    def render_actor(self, value):
+        return "%s" % value.get_display_name()
+
+
+class LogFilter(django_filters.FilterSet):
+    def __init__(self, user=None, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if not self.request.user.has_perm("coop.view"):
+            userfield_choices = [
+                (
+                    TapirUser.objects.get(username=self.request.user).id,
+                    TapirUser.objects.get(
+                        username=self.request.user
+                    ).get_display_name(),
+                )
```

Comment:
self.request.user is already a TapirUser, so you can do something like (haven't tested it) :

```suggestion
                    self.request.user.id,
                    self.request.user.get_display_name(),
```

Reply:
tested it, seems to work